app/index.js

```diff
@@ -147,11 +182,130 @@ module.exports = class extends yeoman {
         userProps.package = this.package;
 
         for (var i = 0; i < this.files.length; i++) {
-            this.fs.copyTpl(
-                this.templatePath(this.files[i]),
-                this.destinationPath(this.files[i].replace(/src\/main\/java/g, path.join(src, packageFolder))),
-                { userProps: userProps }
-            );
+            var skipFile = false;
+            if (showWsdl2Rest) {
+                skipFile = (testFileForWsdl2RestSkip(this.files[i]) == true);
+            } else {
+                if(this.files[i].localeCompare('pom.xml.wsdl2rest') == 0) {
+                    skipFile = true;
+                }
+            }
+
+            if (!skipFile) {
+                if(this.files[i].localeCompare('pom.xml.wsdl2rest') == 0) {
+                    var tempOutFile = "pom.xml";
+                    this.fs.copyTpl(
+                        this.templatePath(this.files[i]),
+                        this.destinationPath(tempOutFile.replace(/src\/main\/java/g, path.join(src, packageFolder))),
+                        { userProps: userProps }
+                    );
+                } else {
+                    this.fs.copyTpl(
+                        this.templatePath(this.files[i]),
+                        this.destinationPath(this.files[i].replace(/src\/main\/java/g, path.join(src, packageFolder))),
+                        { userProps: userProps }
+                    );
+                }
+            }
+        }
+
+        if (showWsdl2Rest) {
+            return wsdl2restGenerate(this.wsdl, this.outdirectory, this.camelDSL, showDebug);
         }
     }
 };
+
+var skipListForWsdl2RestFiles = [
+    'src/main/resources/META-INF/spring/camel-context.xml',
+    'src/main/resources/OSGI-INF/blueprint/blueprint.xml',
+    'pom.xml'
+];
+
+function testFileForWsdl2RestSkip(file) {
+    for(var i=0; i<skipListForWsdl2RestFiles.length; i++) {
+        if (skipListForWsdl2RestFiles[i] == file) return true;
+    }
+}
+
+function wsdl2restGenerate(wsdlUrl, outputDirectory, dsl, isDebug) {
+    if (dsl.indexOf('java') > 0) {
```

Comment:
Eclipse is suggesting me to use include instead of "dsl.indexOf('java') > 0"
it seems more readable, what do you think?